<?xml version="1.0"?>
<launch>

  <arg name="ns" default="o3d3xx_h5"/>
  <arg name="nn" default="h5writer"/>
  <arg name="outdir" default="/tmp"/>
  <arg name="outfile" default="o3d3xx.h5"/>
  <!--
   Sensible modes include:
      'w'  - Create file, truncate if exists
      'w-' - Create file, fail if exists
      'a'  - Read/write, create otherwise
  -->
  <arg name="open_mode" default="w-"/>
  <!--
    GZip compression level: 0 - 9, where 9 is "most compression"
  -->
  <arg name="compression" default="9"/>
  <!--
    Puts a limit on how many images (per type) get written to the HDF5
    file.
  -->
  <arg name="circular_buffer_sz" default="999999999"/>

  <arg name="camera_ns" default="o3d3xx"/>
  <arg name="camera_nn" default="camera"/>
  <arg name="suffix" default=""/>

  <node pkg="o3d3xx_h5"
        type="h5writer"
        ns="$(arg ns)"
        name="$(arg nn)"
        output="screen">

    <!-- Subscribed topics/services -->
    <remap from="/xyz_image"
           to="/$(arg camera_ns)/$(arg camera_nn)/xyz_image$(arg suffix)"/>

    <remap from="/amplitude"
           to="/$(arg camera_ns)/$(arg camera_nn)/amplitude$(arg suffix)"/>

    <remap from="/depth"
           to="/$(arg camera_ns)/$(arg camera_nn)/depth$(arg suffix)"/>

    <remap from="/confidence"
           to="/$(arg camera_ns)/$(arg camera_nn)/confidence$(arg suffix)"/>

    <remap from="/Dump"
           to="/$(arg camera_ns)/$(arg camera_nn)/Dump"/>

    <!-- Runtime parameters -->
    <param name="outdir" value="$(arg outdir)"/>
    <param name="outfile" value="$(arg outfile)"/>
    <param name="open_mode" value="$(arg open_mode)"/>
    <param name="compression" value="$(arg compression)"/>
    <param name="circular_buffer_sz" value="$(arg circular_buffer_sz)"/>

  </node>

</launch>
